tools/create_index/index_generator.js

```diff
@@ -0,0 +1,37 @@
+var fs = require('fs')
+
+// We will fill in this object with our data
+var vuln = {}
+
+const createIndex = function (vulnDirectoryPath) {
+  const files = fs.readdirSync(vulnDirectoryPath)
+  getVulnDirectoryContents(files, vulnDirectoryPath)
+  writeIndex(vuln, vulnDirectoryPath)
+}
+
+const getVulnDirectoryContents = function (files, vulnDirectoryPath) {
+  for(file of files) {
+    const filename = file.slice(-0, file.toString().indexOf('.json'))
+    if(filename !== 'index') {
+      const data = fs.readFileSync(vulnDirectoryPath + file)
+      createVulnObject(filename, data)
+    }
+  }
+}
+
+const createVulnObject = function(identifier, json) {
+  vuln[identifier] = JSON.parse(json)
```

Comment:
What would be the benefit of mirroring the entire report content into the index json?
I would think that we'd really only want a simple index, as in just the identifier itself?

Some difficulties it may raise is:
1. We usually one-off edit a report just to change its score, references or something like this. If we have a mirror of that in the index, it means we need to remember updating both of them, or more ideally have some kind of linter to check that and fail the build if the same report is different than its index copy
2. An overall duplication of data

Reply:
I can get a list of identifiers very easily – that data isn't really useful as someone trying to work with this dataset, as I'll still need to write custom tooling to read all of the files. The intent here is to be able to parse all of the data easily from the source, similar to what we do in the vulnerability-focused parts of [node-release-lines](https://github.com/nw/node-release-lines).

Addressing your points:
- Happy to see if I can add an additional npm script that can be run in CI to compare the current `index.json` to a newly-generated `index.json`. If they're different, `process.exit(1)`. If they're the same, `process.exit(0)`. Will that resolve your concerns around this?
- Since this is automated data generation, I don't see this as an issue – I see it as an improvement to DX. The individual files remain the source of truth that we maintain, this is just an improved way to consume them as a developer who isn't working on the Security WG but does want to work with this data.

Reply:
I'd agree that avoiding duplication of the data is what makes sense to me.

Reply:
Having all the data in a single, usable location is the utility of this work to me. Definitely interested in understanding what the concern is for y'all.